pkg/process/config/yaml_config.go

```diff
@@ -172,6 +172,21 @@ func (a *AgentConfig) loadSysProbeYamlConfig(path string) error {
 		a.Windows.DriverBufferSize = driverBufferSize
 	}
 
+	// If there is no network_config, assume it is an old version of the config, and enable
+	// the network check. This has the downside that if the network_config is deleted, network
+	// check will be enabled.  However, this seems like the best option for backwards compatibility.
+	//
+	// The network check won't be run if sysprobe isn't enabled so there's no need to check
+	// EnableSystemProbe here.
+	networkEnabled := true
+	if config.Datadog.IsSet(key(spNS, "network_config")) {
+		// System probe can be run without the network module as determined on the network_config.enabled value
+		networkEnabled = config.Datadog.GetBool(key(spNS, "network_config.enabled"))
+	}
+	if networkEnabled {
+		a.EnabledChecks = append(a.EnabledChecks, "Network")
```

Comment:
I think we want to have separate config fields for the enabled process-agent checks (which this field refers to) and the enabled system-probe modules. 

It'll get confusing if this field is both a list of both.  My suggestion is to add a field to config called `EnabledModules`

Reply:
EnabledChecks is used for OOM Kill and TCP queue length as well, which are system probe checks, which is why I included Network in it.

Should I still change it?  Or should I still add EnabledModules, but add OOM and TCP to that list?  Or add EnabledModules and only add Network to it?

Reply:
(resolved on slack:
Lee:  In that case, I think it’s fine to keep it as is. It’s a little weird that this is controlling both process checks and  sysprobe modules, but I suppose we’re already doing that)